llrt_core/src/vm.rs

```diff
@@ -697,3 +705,34 @@ fn set_import_meta(module: &Module<'_>, filepath: &str) -> Result<()> {
     meta.prop("url", ["file://", filepath].concat())?;
     Ok(())
 }
+
+pub fn get_module_path_in_node_modules(ctx: &Ctx<'_>, specifier: &str) -> Result<String> {
+    let ex = Exception::throw_reference(ctx, &format!("Error resolving module '{}'", specifier));
+    let current_dir = env::current_dir()
+        .map(|path| path.display().to_string())
+        .unwrap_or_default();
+
+    let (mut scope, mut name) = match specifier.split_once('/') {
+        Some((s, n)) => (s, ["./", n].concat()),
+        None => (specifier, ".".to_string()),
+    };
+
+    let mut package_json_path = [&current_dir, "/node_modules/", scope, "/package.json"].concat();
+    if !Path::new(&package_json_path).exists() && name != "." {
+        scope = specifier;
+        name = ".".to_string();
+        package_json_path = [&current_dir, "/node_modules/", scope, "/package.json"].concat();
```

Comment:
Maybe we can do something like this instead:
```rust
let (scope, name) = specifier.split_once('/').unwrap_or((specifier, "."));
let mut package_json_path = [&current_dir,"/node_modules/"].concat();
let base_path_length = package_json_path.len();
package_json_path.push_str(scope);
package_json_path.push_str("/package.json");

let (scope, name) = if name != "." && !fs::metadata(&package_json_path).is_ok() { //check name first to avoid syscall
    package_json_path.truncate(base_path_length); //keep string
    package_json_path.push_str(specifier);
    package_json_path.push_str("/package.json");
    (specifier, ".")
} else {
    (scope, name)
};

if fs::metadata(&package_json_path).is_err() {
   //TODO handle error if doesn't exist 
}

let package_json = fs::read(&package_json_path).unwrap_or_default();
let package_json: JsonValue = simd_json::to_owned_value(&package_json)
  ```

Reply:
Excellent. You have replaced what I had in mind with a clearer implementation. :)

Reply:
The only thing I wish I could have done to implement this is that if the scope and name were separated, the name would be “. /foo” format.

```rust
let (scope, name) = specifier.split_once('/').unwrap_or((specifier, "."));
```